### ISSUE TITLE
JupyterHub container size selector takes a long time to display (osc-cl1/osc-cl2)

### ISSUE THREAD

**Issue by MichaelTiemannOSC** [operate-first/apps#1918]
There are several comments in https://github.com/operate-first/apps/issues/1748 saying that the problem is not yet fixed.  It's not yet fixed:
![Screen Shot 2022-04-25 at 6 01 24 AM](https://user-images.githubusercontent.com/72577720/165067779-11b64e1f-f560-4c54-b94e-d0c6f4339d01.png)

There should be no delay when offering the Deployment Size option to start a notebook.  Presently, in the best of times, the delay is a mere few seconds.  Most of the time, however, it takes infinitely long, which means the option is not practically available without refreshing, restarting, reloading, some random things until it "works".

@erikerlandson @MichaelClifford

**Comment by HumairAK** [comment]
Seems like at times the container sizes do not display at all. Looking at the console logs error on the front end, we can see: 

![image](https://user-images.githubusercontent.com/10904967/165374343-af5f5a30-f968-425a-b4d7-5bfa7e8a0ea8.png)

**Comment by HumairAK** [comment]
See gif for how this looks like: 
![JH container size delay Peek 2022-04-26 15-10](https://user-images.githubusercontent.com/10904967/165374983-29c1885e-6c9e-4eff-8e2a-00fd9c11ac83.gif)

The first attempt it seems successful, in the second one you can see the error displayed at which point no "container size" shows up.

**Comment by MichaelTiemannOSC** [comment]
I'm seeing similar things:
![Screen Shot 2022-04-27 at 8 59 35 AM](https://user-images.githubusercontent.com/72577720/165524882-a8a7ac3f-2bda-4c0f-8624-cb3ba79c9841.png)

**Comment by HumairAK** [comment]
@lucferbux or @maroroman do you folks have a suggestion on a workaround here? or things we can attempt to resolve this?

**Comment by MichaelTiemannOSC** [comment]
I think this is fixed by https://github.com/os-climate/os_c_data_commons/issues/162

**Comment by HumairAK** [comment]
@MichaelTiemannOSC that seems like a different issue, but I suppose if the problem is gone then we no longer need this, otherwise please re-open it if you encounter it again.

**Comment by MichaelTiemannOSC** [comment]
This issue is back and worse than ever.

**Comment by MichaelTiemannOSC** [comment]
This morning the issue in CL2 seems fixed--the lag to display the size option (which was remembered from my previous day's work) was less than 1 second.  Thanks!  Will close if the fix sticks.

**Comment by MichaelTiemannOSC** [comment]
I spoke too soon: the 32GB configuration that was presented was a lie--the notebook started with 2GB, which then crashed when more memory was required.  I'm now trying to resstart and get a proper 32GB notebook.  But I'm now waiting minutes for it to come back with options I can choose.

**Comment by MichaelTiemannOSC** [comment]
I'm looking at the two YAML files that populate these lists, and notice a difference:

https://github.com/operate-first/apps/blob/master/kfdefs/overlays/osc/osc-cl2/jupyterhub/profiles/singleuser-profiles-container-sizes.yaml begins:

```
kind: ConfigMap
apiVersion: v1
metadata:
  name: singleuser-profiles-container-sizes
  labels:
    jupyterhub: singleuser-profiles
data:
  jupyterhub-singleuser-profiles.yaml: |
    sizes:
    - name: Medium
      resources:
        requests:
          memory: 2Gi
          cpu: 100m
        limits:
          memory: 4Gi
          cpu: 2
    [...truncated...]
```

https://github.com/operate-first/apps/blob/master/kfdefs/overlays/osc/osc-cl2/jupyterhub/profiles/singleuser-profiles-gpu-ui.yaml contains:

```
kind: ConfigMap
apiVersion: v1
metadata:
  name: singleuser-profiles-gpu-ui
  labels:
    jupyterhub: singleuser-profiles
data:
  jupyterhub-singleuser-profiles.yaml: |
    ui:
      gpuConfig:
        enabled: True
        type: dropdown
        gpuDropdown:
            start: 0
            end: 4
```

Would it help to use the `ui:` and some sort of radiobutton stanza to just get the size options rendered, rather than having text that should, ultimately, be rendered as a set of radio buttons after some nondeterministic amount of time and trouble?

**Comment by DaBeIDS** [comment]
I had the same problem as Michael mentioned before. After 20-30 seconds the container size appears.

**Comment by MichaelTiemannOSC** [comment]
I have a new theory: if the jupyterhub spawner is "cold" (if I'm the first person to wake it up for the day), it may actually take until somebody else starts a notebook before the container size appears.  When I did a cold start this morning, I started the default (2GB) notebook, shut it down immediately after it successfully spawned, then I went back to the hub to start again.  I could see the container size option pop up then within seconds.  So perhaps the problem is that the container size feature present in the OSC overlay is relying on a side-effect of the base jupyterhub spawner, and without that side-effect, it's not being activated correctly.
